cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -677,6 +678,185 @@ def _validate_dataframe(self, df_name: str):
                         self.warnings.append(column_def["warning_message"])
                     self._validate_column(column, column_name, df_name, column_def)
 
+    def _validate_colors_in_uns_dict(self, uns_dict: dict) -> None:
+        df = getattr_anndata(self.adata, "obs")
+        df_definition = self._get_component_def("obs")
+
+        # Mapping from obs column name to number of unique categorical values
+        category_mapping = {}
+
+        if "columns" in df_definition:
+            for column_name, column_def in df_definition["columns"].items():
+                if column_name not in df.columns:
+                    # Skip this, dataframe validation should already append an error for this
+                    continue
+
+                if column_def.get("type") == "categorical":
+                    category_mapping[column_name] = df[column_name].nunique()
+
+        for column_name, num_unique_vals in category_mapping.items():
+            colors_options = uns_dict.get(f"{column_name}_colors", [])
```

Comment:
We should only run validation against {column_name}_colors if it exists, as having a corresponding {column}_colors column is optional. As written, this will fail if a categorical column opts out of {column}_colors